### PR TITLE
feat: Disable FS backend when user provides one

### PR DESCRIPTION
**Pull request by filipesmedeiros**
One thing to note is that it seemed necessary to make the changes to the way we handle the passing of the config to the client.
Next cannot serialize stuff like functions, so we cannot return everything and assume it's serializable, especially the `use` array and the `backend` prop (with the props inside).
The way this was done now requires the user to pass the same config to `appWithTranslation` that they pass to `serverSideTranslations`. This is kind of repetitive and limiting (same client side config for the whole app). We could allow the user the to pass a function that decides what config to use based on the current page, but that seems troublesome.

Let me know what you think!

### CODE REVIEW

**Comment by LeonardDrs** [comment]
Tried this modification on my app, it works really well.
Dunno if you've seen my edit to this [comment](https://github.com/isaachinman/next-i18next/issues/931#issuecomment-772686669) but I didn't had to pass the configuration myself to `serverSideTranslations` and the `sprintf-postProcessor` worked just fine in SSG/SSR

**Comment by filipesmedeiros** [comment]
Your comment, doesn't it mean you gotta have `next-i18next.config.js` inside `src`? Or was it outside?

**Comment by LeonardDrs** [comment]
Nop, `next-i18next.config.js` is right beside `package.json` and `next.config.js` on the root of the project

**Comment by filipesmedeiros** [comment]
I will rebase this is the branch with the new changes @isaachinman just merged

**Comment by isaachinman** [comment]
Sounds good, thanks!

File: src/createClient/node.ts [hunk 1]
```diff
@@ -8,7 +8,9 @@ export default (config: InternalConfig): CreateClientReturn => {
   let initPromise: InitPromise
 
   if (!instance.isInitialized) {
-    instance.use(i18nextFSBackend)
+    const hasCustomBackend = config?.use?.some((b) => b.type === 'backend')
+    if (!hasCustomBackend)
```
Comment (isaachinman) [inline src/createClient/node.ts:12]: Let's use curly braces here, please.

File: types.d.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export type UserConfig = {
   use?: any[]
 } & InitOptions
 
+
```
Comment (isaachinman) [inline types.d.ts:26]: Shouldn't be necessary.

**Review by isaachinman — CHANGES_REQUESTED** [review]

File: src/createClient/node.ts (hunk 1 above)
Comment (filipesmedeiros) [reply src/createClient/node.ts]: Sure thing!

File: types.d.ts (hunk 2 above)
Comment (filipesmedeiros) [reply types.d.ts]: My bad, removed

**Review by isaachinman — APPROVED** [review]

**Pull request merged**